lib/datasource.js

```diff
@@ -1450,17 +1452,21 @@ function capitalize(str) {
   return str.charAt(0).toUpperCase() + ((str.length > 1) ? str.slice(1).toLowerCase() : '');
 }
 
-function fromDBName(dbName, camelCase) {
-  if (!dbName) {
-    return dbName;
-  }
-  const parts = dbName.split(/-|_/);
-  parts[0] = camelCase ? parts[0].toLowerCase() : capitalize(parts[0]);
-
-  for (let i = 1; i < parts.length; i++) {
-    parts[i] = capitalize(parts[i]);
+/**
+ * Renames db column names with different naming conventions:
+ * camelCase for property names as it's LB default naming convention for properties,
+ * or keep the name the same if needed.
+ *
+ * @param {*} colName column name defined in database
+ * @param {*} defcamelCase boolean. Converts the name to camelcase if true as
+ * it's the default case for variables / properties in LoopBack.
+ */
+function fromDBName(colName, defCamelCase) {
```

Comment:
It's probably better to make 2nd argument an optional function (default to camelCase) for case conversion. Using a `boolean` argument is not very friendly for JS.

- fromDBName(colName) or fromDBName(colName, camelCase) // convert to camel case
- fromDBName(colName, noCase) // no conversion
- fromDBName(colName, pascalCase)   // convert to pascal case